component/config_forwarding.libsonnet

```diff
@@ -117,69 +137,167 @@ local patchCustomPipelines = {
+  ],
+  [if std.length(specs.pipelines) > 0 then 'pipelines']: [
+    { name: name } + specs.pipelines[name]
+    for name in std.objectFields(specs.pipelines)
+  ],
+} + {
+  // Import remaining specs as is.
+  [key]: specs[key]
+  for key in std.objectFields(specs)
+  if !std.member([ 'inputs', 'outputs', 'pipelines' ], key)
+};
 
 // ClusterLogForwarder:
 // Create definitive ClusterLogForwarder resource from specs.
 local clusterLogForwarder = lib.ClusterLogForwarder(params.namespace, 'instance') {
-  spec: {
-    // Unfold objects into array.
-    [if std.length(clusterLogForwarderSpec.inputs) > 0 then 'inputs']: [
-      { name: name } + clusterLogForwarderSpec.inputs[name]
-      for name in std.objectFields(clusterLogForwarderSpec.inputs)
-    ],
-    [if std.length(clusterLogForwarderSpec.outputs) > 0 then 'outputs']: [
-      { name: name } + clusterLogForwarderSpec.outputs[name]
-      for name in std.objectFields(clusterLogForwarderSpec.outputs)
-    ],
-    [if std.length(clusterLogForwarderSpec.pipelines) > 0 then 'pipelines']: [
-      { name: name } + clusterLogForwarderSpec.pipelines[name]
-      for name in std.objectFields(clusterLogForwarderSpec.pipelines)
-    ],
-  } + {
-    // Import remaining specs as is.
-    [key]: clusterLogForwarderSpec[key]
-    for key in std.objectFields(clusterLogForwarderSpec)
-    if !std.member([ 'inputs', 'outputs', 'pipelines' ], key)
-  },
+  spec: unfoldSpecs(clusterLogForwarderSpec),
 };
 
+// namespaceLogForwarder:
+// Create namespaced LogForwarder resource from specs.
+local namespaceLogForwarder = [
+  local specs = { inputs: {}, outputs: {}, pipelines: {} } + com.makeMergeable(params.namespaceLogForwarder[forwarder]);
+  local name = utils.namespacedName(forwarder).name;
+  local namespace = utils.namespacedName(forwarder).namespace;
+  local serviceAccount = std.get(specs, 'serviceAccountName', utils.namespacedName(forwarder).name);
+
+  lib.ClusterLogForwarder(namespace, name) {
+    spec: { serviceAccountName: serviceAccount } + com.makeMergeable(unfoldSpecs(specs)),
+  }
+  for forwarder in std.objectFields(params.namespaceLogForwarder)
+  if !std.member([ 'enabled', 'instance', 'openshift-logging/instance' ], forwarder)
+];
+
+// namespaceServiceAccount:
+// Create ServiceAccount for namespaced LogForwarder specs.
+local namespaceServiceAccount = [
+  local specs = params.namespaceLogForwarder[forwarder];
+  local namespace = utils.namespacedName(forwarder).namespace;
+  local serviceAccount = std.get(specs, 'serviceAccountName', utils.namespacedName(forwarder).name);
+
+  if serviceAccount == '' then std.trace('Warning! No ServiceAccountName set in namespaceLogForwarder!', {})
+  else kube.ServiceAccount(serviceAccount) {
+    metadata+: {
+      namespace: namespace,
+    },
+    spec: {
+
+    },
+  }
+  for forwarder in std.objectFields(params.namespaceLogForwarder)
+  if !std.member([ 'enabled', 'instance', 'openshift-logging/instance' ], forwarder)
```

Comment:
Especially since you have the same list multiple times. If you wanna keep it like this, at least have the excluded fields in a single place as a local variable.

Reply:
I will do that, I think I wanna keep it that way; This gives us the option to remove the `enabled` parameter in a future release and keep this parameter as is 🤔 maybe I could move the `enabled` into a seperate "top-level" toggle.